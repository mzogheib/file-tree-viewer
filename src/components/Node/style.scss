.node {
  margin: 5px 0;

  &__child {
    padding: 0 0 0 20px;
  }
}
