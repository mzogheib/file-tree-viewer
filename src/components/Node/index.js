import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './style.scss'
import NodeFile from '../NodeFile'
import NodeFolder from '../NodeFolder'

class Node extends Component {
  constructor() {
    super()
    this.state = {
      open: false,
    }
    this.toggleOpen = this.toggleOpen.bind(this)
  }

  toggleOpen(event) {
    event.stopPropagation()
    if (this.props.node.type === 'file') {
      return
    }
    this.setState({ open: !this.state.open })
  }

  renderNode(node) {
    const canRenderChildren =
      this.state.open && node.children && !!node.children.length
    return (
      <div className="node" onClick={this.toggleOpen}>
        {node.type === 'file' && <NodeFile node={node} />}
        {node.type === 'folder' && (
          <NodeFolder node={node} open={this.state.open} />
        )}
        {canRenderChildren &&
          node.children.map((childNode, index) => (
            <div key={index} className="node__child">
              <Node node={childNode} />
            </div>
          ))}
      </div>
    )
  }

  render() {
    return this.renderNode(this.props.node, this.props.depth)
  }
}

export default Node

Node.propTypes = {
  node: PropTypes.object.isRequired,
}
