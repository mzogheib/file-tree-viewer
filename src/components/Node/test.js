import React from 'react'
import renderer from 'react-test-renderer'
import Node from './'

const mockResponse = {
  data: [
    {
      type: 'file',
      name: 'auto_loan_account.png',
      size: 67002,
    },
    {
      type: 'folder',
      name: 'redundant',
      children: [
        {
          type: 'file',
          name: 'toys_matrix.wav',
          size: 26676,
        },
        {
          type: 'file',
          name: 'optical_withdrawal.pdf',
          size: 2272,
        },
        {
          type: 'folder',
          name: 'open-source',
          children: [],
        },
        {
          type: 'file',
          name: 'implement_hack.html',
          size: 19929,
        },
        {
          type: 'file',
          name: 'index_overpass_generic.mpg',
          size: 94248,
        },
      ],
    },
    {
      type: 'folder',
      name: 'redundant',
      children: [
        {
          type: 'file',
          name: 'system_worthy.shtml',
          size: 38689,
        },
        {
          type: 'file',
          name: 'reduced_incredible_wooden_bike_monitor.html',
          size: 60805,
        },
      ],
    },
    {
      type: 'file',
      name: 'italy_input_avon.mp4v',
      size: 3119,
    },
    {
      type: 'file',
      name: 'neural.shtml',
      size: 87349,
    },
  ],
}

const makeTree = node => renderer.create(<Node node={node} />).toJSON()

// Snapshot test to ensure correct icons and filesize is displayed
it('renders correctly', () => {
  mockResponse.data.forEach(node => expect(makeTree(node)).toMatchSnapshot())
})
