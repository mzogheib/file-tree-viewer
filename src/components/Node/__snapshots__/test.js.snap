// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="node"
  onClick={[Function]}
>
  <div
    className="node-file"
  >
    <div
      className="node-file__icon undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
      >
        <path
          d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
        />
      </svg>
    </div>
    <span
      className="node-file__name"
      title="auto_loan_account.png"
    >
      auto_loan_account.png
    </span>
    <span
      className="node-file__size"
    >
      65 KB
    </span>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="node"
  onClick={[Function]}
>
  <div
    className="node-folder"
  >
    <div
      className="node-folder__chevron undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
        />
      </svg>
    </div>
    <div
      className="node-folder__icon undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
        />
      </svg>
    </div>
    <span
      className="node-folder__name"
      title="redundant"
    >
      redundant
    </span>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="node"
  onClick={[Function]}
>
  <div
    className="node-folder"
  >
    <div
      className="node-folder__chevron undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 320 512"
        width="1em"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
        />
      </svg>
    </div>
    <div
      className="node-folder__icon undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
        />
      </svg>
    </div>
    <span
      className="node-folder__name"
      title="redundant"
    >
      redundant
    </span>
  </div>
</div>
`;

exports[`renders correctly 4`] = `
<div
  className="node"
  onClick={[Function]}
>
  <div
    className="node-file"
  >
    <div
      className="node-file__icon undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
      >
        <path
          d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
        />
      </svg>
    </div>
    <span
      className="node-file__name"
      title="italy_input_avon.mp4v"
    >
      italy_input_avon.mp4v
    </span>
    <span
      className="node-file__size"
    >
      3 KB
    </span>
  </div>
</div>
`;

exports[`renders correctly 5`] = `
<div
  className="node"
  onClick={[Function]}
>
  <div
    className="node-file"
  >
    <div
      className="node-file__icon undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 384 512"
        width="1em"
      >
        <path
          d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
        />
      </svg>
    </div>
    <span
      className="node-file__name"
      title="neural.shtml"
    >
      neural.shtml
    </span>
    <span
      className="node-file__size"
    >
      85 KB
    </span>
  </div>
</div>
`;
